### DISCUSSION TITLE
Automerge "sometimes" does not work

### DISCUSSION THREAD

**Discussion by keita-makino** [renovatebot/renovate#9437] [Request Help] [answered]
I have the following settings in one of my repos but *sometimes* automerge is not done by the bot and I need to merge it manually. When does it happen and how to address it?

### Settings

```json
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "pin", "digest"],
      "automerge": true
    }
  ]
}
```

### Does automerge
![image](https://user-images.githubusercontent.com/26666870/113941305-b4037e00-97b3-11eb-8445-4bad0fc69e56.png)

### Does not automerge
![image](https://user-images.githubusercontent.com/26666870/113941262-9cc49080-97b3-11eb-816d-bdd467f02e4d.png)

**Comment by viceice** [comment, marked as answer]
Renovate needs at least two runs to automerge. In first run it creates pr and in second run (normally one hour later) it tries to automerge.

**Reply by keita-makino** [reply]
But there are many PRs that have been left as is for many days, what about them?
Example (my repo): https://github.com/keita-makino/qualtrics-map/pulls

**Reply by viceice** [reply]
Please provide the debug logs, to check why they aren't merged.

**Reply by rarkins** [reply]
What do the logs from the last run indicate? Eg did it try to automerge at least one branch?

**Reply by keita-makino** [reply]
https://pastebin.com/VBwYF0Tx
Is this what you need to look at?

**Reply by viceice** [reply]
Branch status pending, you have recently changed required status checks. Those prs have a missing required status check.

**Reply by keita-makino** [reply]
So what should I do to let it try automerging again?

**Reply by viceice** [reply]
Validate your branch protections, so that all required status checks are running on push / pr

**Reply by keita-makino** [reply]
Indeed I have no branch protections right now.

![image](https://user-images.githubusercontent.com/42898720/114081003-e4552600-9860-11eb-9b76-c8c8cd6f7961.png)

**Reply by viceice** [reply]
Strange, if you check your logs you see pending state 🤔 maybe you changed that recently?

**Reply by keita-makino** [reply]
I believe I've never set a branch protection in this repo... 🤔

**Reply by viceice** [reply]
```
DEBUG: Checking if we can automerge branch(branch="renovate/babel-monorepo")
DEBUG: mergeStatus=no automerge(branch="renovate/babel-monorepo")
DEBUG: Ensuring PR(branch="renovate/babel-monorepo")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/babel-monorepo")
DEBUG: Found existing PR(branch="renovate/babel-monorepo")
DEBUG: Removing duplicate release notes(branch="renovate/babel-monorepo")
{
  "depName": "@babel/core"
}
DEBUG: Removing duplicate release notes(branch="renovate/babel-monorepo")
{
  "depName": "@babel/preset-react"
}
DEBUG: getBranchStatus(renovate/babel-monorepo)(branch="renovate/babel-monorepo")
DEBUG: branch status check result(branch="renovate/babel-monorepo")
{
  "state": "pending",
  "statuses": []
}
DEBUG: No check runs found(branch="renovate/babel-monorepo")
{
  "result": {
    "total_count": 0,
    "check_runs": []
  }
}
```

No status checks, so no automerge. Renovate will by default never automerge without any checks

**Reply by keita-makino** [reply]
What kind of settings (checks) should be added there then?
Also, it doesn't explain why some of the PRs would automerge and some others wouldn't. If a repo setting affect the behavior all the PR must behave consistently, isn't it?

**Reply by viceice** [reply]
I don't think any pr was automerged on your https://github.com/keita-makino/qualtrics-map repo because you don't have any kind of ci checks configured.

We highly recommend to have a ci validation before automerging any update, otherwise renovate can break your main branch!

**WARNING! HIGH RISK**
If you really want to automerge without ci check, use [requiredStatusChecks](https://docs.renovatebot.com/configuration-options/#requiredstatuschecks).

**Reply by keita-makino** [reply]
Hmm but then why some of the PRs are automerged without any settings? Now I understand checks are needed for the feature but my point is why there's an inconsistency among the PRs.

**Reply by rarkins** [reply]
Can you point to any PR which was automerged? I couldn't see any

**Reply by keita-makino** [reply]
Oh I see, I thought the autoclosed cases are automerged but they have been closed as I manually updated the packages and pushed the commit.
Thank you for your help, I will add a ci to enable the feature.
